lib/tizen_emulator.dart

```diff
@@ -272,42 +285,70 @@ class TizenEmulators extends EmulatorDiscovery {
     properties['profile'] = profile;
     properties['version'] = version;
 
-    return TizenEmulator(id,
-        properties: properties,
-        logger: _logger,
-        processManager: _processManager);
+    return TizenEmulator(
+      id,
+      properties: properties,
+      logger: _logger,
+      processManager: _processManager,
+      tizenSdk: _tizenSdk,
+    );
   }
 
+  /// Tizen emulator supports all major platforms(Windows/macOS/Linux).
   @override
-  // TODO(HakkyuKim): fix the code so that it actually evaluates the bool value
   bool get supportsPlatform => true;
 }
 
 class TizenEmulator extends Emulator {
-  TizenEmulator(String id,
-      {Map<String, String> properties,
-      @required Logger logger,
-      @required ProcessManager processManager})
-      : _properties = properties,
-        _processUtils =
-            ProcessUtils(logger: logger, processManager: processManager),
-        super(id, properties != null && properties.isNotEmpty);
+  TizenEmulator(
+    String id, {
+    Map<String, String> properties,
+    @required Logger logger,
+    @required ProcessManager processManager,
+    @required TizenSdk tizenSdk,
+  })  : _logger = logger,
+        _properties = properties,
+        _processUtils = ProcessUtils(
+          logger: logger,
+          processManager: processManager,
+        ),
+        _tizenSdk = tizenSdk,
+        super(
+          id,
+          properties != null && properties.isNotEmpty,
+        );
 
+  final Logger _logger;
   final Map<String, String> _properties;
   final ProcessUtils _processUtils;
+  final TizenSdk _tizenSdk;
 
   String _prop(String name) => _properties != null ? _properties[name] : null;
 
-  // Should we subcategorize it based on Tizen profile?
+  // TODO(HakkyuKim): Consider subcategorizing into Tizen profiles.
   @override
   Category get category => Category.mobile;
 
+  /// See [AndroidEmulator.launch] in [android_emulator.dart] (simplified)
   @override
   Future<void> launch() async {
-    // TODO(HakkyuKim): launch is overly simplified, for example, it currently doesn't record the emulator process.
-    /// See [AndroidEmulator.launch()] in [android_emulator.dart]
-    final List<String> args = <String>[getEmCliPath(), 'launch', '--name', id];
-    await _processUtils.run(args);
+    // Null checking on _tizenSdk.emCli.path is already
+    // done at `TizenEmulators._getEmulators`.
```

Comment:
It is not very clear that `TizenEmulators._getEmulators` is always invoked before `TizenEmulator.launch` is called.

By the way I think this:
```dart
  @override
  Future<List<Emulator>> get emulators => _getEmulators();

  Future<List<Emulator>> _getEmulators() async {
```
can be just
```dart
  @override
  Future<List<Emulator>> get emulators async {
```

Reply:
`TizenEmulator` class cannot access `TizenEmulators.canLaunchAnything`. I can't think of a good solution at the moment, should I repeat the code like this?

```dart
Future<void> launch() async {
  if(!(_tizenSdk?.emCli?.existsSync() ?? false)){
    throwToolExit('Unable to locate Tizen Emulator Manager.');
  }

  ...
```

Reply:
No need to call `canLaunchAnything`. It's just a null check.

Something like this:
```dart
Future<void> launch() async {
  final String emCliPath = _tizenSdk?.emCli?.path;
  if (emCliPath == null) {
    throwToolExit('Unable to locate Tizen Emulator Manager.');
  }
  final List<String> command = <String>[
      emCliPath,
      ...
```